crates/core/src/client/mod.rs

```diff
@@ -49,11 +49,11 @@ use crate::models::event::StarknetEvent;
 use crate::models::felt::Felt252Wrapper;
 use crate::models::transaction::{StarknetTransaction, StarknetTransactions};
 
-pub struct KakarotClient<StarknetClient>
+pub struct KakarotClient<StarknetProvider>
```

Comment:
I've always found this line weird, usually you would use T for the generic param. Propose to rewrite to:
```
pub struct KakarotClient<T: Provider>
```